queries.sql

```diff
@@ -7,4 +7,57 @@ SELECT date_of_birth from animals WHERE name='Agumon' OR name='Pikachu';
 SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
-SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
\ No newline at end of file
+SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
+
+begin;
+UPDATE animals SET species = 'unspecified';
+rollback;
+select * from animals;
+
+begin
+update animals set species = 'digimon' where name like '%mon';
+UPDATE animals set species = 'pokemon' WHERE name not like '%mon';
+commit
+select * from animals
+
+begin
+delete from animals
+rollback
+select * from animals
+
+begin
+delete from animals where date_of_birth > '2022-1-1'
+savepoint check1
+select * from animals
+update animals set weight_kg = weight_kg * -1
+select * from animals
+rollback to check1
+select * from animals
+update animals set weight_kg = weight_kg * -1 where weight_kg < 0 ;
+select * from animals
+commit
+select * from animals
+
+SELECT count(*) FROM animals;
+
+select * from animals;
+
+SELECT count(*) FROM animals WHERE escape_attempts = 0;
+
+select * from animals;
+
+SELECT avg(weight_kg) FROM animals;
+
+select * from animals;
+
+SELECT max(escape_attempts),neutered FROM animals WHERE neutered = true OR neutered = false GROUP BY neutered ORDER BY max(escape_attempts) DESC;
+
+select * from animals;
+
+SELECT max(weight_kg),min(weight_kg) FROM animals;
```

Comment:
Please correct your request.
As asked by the requirement, we want to find the minimum and maximum weight of each type of animal, so you have to show and group by the species 

```sql
   SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
```